src/services/completions.ts

```diff
@@ -744,8 +744,46 @@ namespace ts.Completions {
             }
         }
 
-        const kind = SymbolDisplay.getSymbolKind(typeChecker, symbol, location);
-        if (kind === ScriptElementKind.jsxAttribute && preferences.includeCompletionsWithSnippetText && preferences.jsxAttributeCompletionStyle && preferences.jsxAttributeCompletionStyle !== "none") {
+        const isJSXAttributeCompletion = contextToken && forEachAncestor(contextToken, (n) => {
+            if (isJsxAttributeLike(n)) {
+                return true;
+            }
+
+            if (isJsxFragment(n) || isJsxOpeningFragment(n) || isJsxClosingFragment(n)) {
+                return false;
+            }
+
+            if (isJsxOpeningElement(n) || isJsxSelfClosingElement(n) || isJsxClosingElement(n)) {

```

Comment:
Should `isJsxClosingElement` be included here? You can't have attributes in a closing element, right?

Reply:
You can't, but I don't want to proceed up the tree any higher than the closest element; the user could totally put a completion at the closing tag.

Feasibly, I could remove the check here, and at worst it'd walk up the tree further than it needs, and do the same thing, though.